xmtp_proto/src/error.rs

```diff
@@ -1,147 +1,258 @@
+                super::Code::Unavailable => tonic::Code::Unavailable,
+                super::Code::DataLoss => tonic::Code::DataLoss,
+                super::Code::Unauthenticated => tonic::Code::Unauthenticated,
+            }
+        }
     }
-}
 
-impl From<InternalError> for Error {
-    fn from(internal: InternalError) -> Self {
-        Error::new(ErrorKind::InternalError(internal))
+    impl From<tonic::Code> for super::Code {
+        fn from(v: tonic::Code) -> super::Code {
+            match v {
+                tonic::Code::Ok => super::Code::Ok,
+                tonic::Code::Cancelled => super::Code::Cancelled,
+                tonic::Code::Unknown => super::Code::Unknown,
+                tonic::Code::InvalidArgument => super::Code::InvalidArgument,
+                tonic::Code::DeadlineExceeded => super::Code::DeadlineExceeded,
+                tonic::Code::NotFound => super::Code::NotFound,
+                tonic::Code::AlreadyExists => super::Code::AlreadyExists,
+                tonic::Code::PermissionDenied => super::Code::PermissionDenied,
+                tonic::Code::ResourceExhausted => super::Code::ResourceExhausted,
+                tonic::Code::FailedPrecondition => super::Code::FailedPrecondition,
+                tonic::Code::Aborted => super::Code::Aborted,
+                tonic::Code::OutOfRange => super::Code::OutOfRange,
+                tonic::Code::Unimplemented => super::Code::Unimplemented,
+                tonic::Code::Internal => super::Code::Internal,
+                tonic::Code::Unavailable => super::Code::Unavailable,
+                tonic::Code::DataLoss => super::Code::DataLoss,
+                tonic::Code::Unauthenticated => super::Code::Unauthenticated,
+            }
+        }
     }
 }
 
-impl fmt::Debug for Error {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let mut f = f.debug_tuple("xmtp::error::Error");
-
-        f.field(&self.kind);
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ApiEndpoint {
+    Publish,
+    SubscribeGroupMessages,
+    SubscribeWelcomes,
+    UploadKeyPackage,
+    FetchKeyPackages,
+    SendGroupMessages,
+    SendWelcomeMessages,
+    QueryGroupMessages,
+    QueryWelcomeMessages,
+    PublishIdentityUpdate,
+    GetInboxIds,
+    GetIdentityUpdatesV2,
+    VerifyScwSignature,
+    QueryV4Envelopes,
+    PublishEnvelopes,
+}
 
-        if let Some(source) = &self.source {
-            f.field(source);
+impl std::fmt::Display for ApiEndpoint {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        use ApiEndpoint::*;
+        match self {
+            Publish => write!(f, "publish"),
+            SubscribeGroupMessages => write!(f, "subscribe_group_messages"),
+            SubscribeWelcomes => write!(f, "subscribe_welcomes"),
+            UploadKeyPackage => write!(f, "upload_key_package"),
+            FetchKeyPackages => write!(f, "fetch_key_package"),
```

Comment:
nit: to make it aligned with others
```suggestion
            FetchKeyPackages => write!(f, "fetch_key_packages"),
```